agent/config.go

```diff
@@ -103,42 +98,49 @@ func (a Agent) identifyCurrentHost() error {
 	if err != nil {
 		return agentError(err)
 	}
-	log.Println("6")
+	log.Println("Retrieved hosts list, found", len(hosts), "hosts")
 
-	// Walking through all interfaces on a host and looking for a
-	// matching interface address in configuration.
-	addrs, _ := net.InterfaceAddrs()
-	for i := range addrs {
-		romanaIP, _, err := net.ParseCIDR(addrs[i].String())
+	// find our host in the list of hosts
+	// checking for a matching IP address
+	// to get the romana CIDR
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return err
+	}
+
+	log.Println("Searching", len(addrs), "interfaces for a matching host configuration")
+
+	// Now find an interface that matches a Romana CIDR
+	// and store that interface's IP address.
+	// It will be used when configuring iptables and routes to tap interfaces.
```

Comment:
Yes, or if the interface name or IP address was provided directly by topology. (Or a command-line option, etc etc.)